### PR TITLE
Ready For new season

### PR DESCRIPTION
**Pull request by MoonMoon2**
Never finished Grade function, but everything else is here.

### CODE REVIEW

File: data.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
 </head>
 
 <body>
-    <table>
+    <table id="t">
```
Comment (ErikBoesen) [inline data.html]: This ID is unnecessary. In JavaScript you can just use `document.getElementsByTagName('table')[0]`.

File: img/ropeclimb-red.svg [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<svg width="125.5" height="110" xmlns="http://www.w3.org/2000/svg">
+ <!-- Created with Method Draw - http://github.com/duopixel/Method-Draw/ -->
```
Comment (ErikBoesen) [inline img/ropeclimb-red.svg]: Lots of unnecessary XML here.

File: img/ropeclimb-red.svg [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <line x1="57.25" y1="37" x2="67.25" y2="47" stroke-width="2" stroke="#ff0000"/>
+    <line x1="57.25" y1="47" x2="67.25" y2="57" stroke-width="2" stroke="#ff0000"/>
+    <line x1="57.25" y1="57" x2="67.25" y2="67" stroke-width="2" stroke="#ff0000"/>
+    <line x1="57.25" y1="67" x2="67.25" y2="77" stroke-width="2" stroke="#ff0000"/>
+    <line x1="57.25" y1="77" x2="67.25" y2="87" stroke-width="2" stroke="#ff0000"/>
+    <line x1="57.25" y1="87" x2="67.25" y2="97" stroke-width="2" stroke="#ff0000"/>
+  </g>
+  <rect rx="6" id="svg_2" height="90" width="10" y="7" x="57.25" stroke-width="5" stroke="#ff0000" fill="none"/>
+ </g>
+</svg>
```
Comment (ErikBoesen) [inline img/ropeclimb-red.svg]: Why do we need a whole duplicate image?

File: img/ropeclimb.svg [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ <g>
+  <title>background</title>
+  <rect fill="none" id="canvas_background" height="402" width="602" y="-1" x="-1"/>
+  <g display="none" overflow="visible" y="0" x="0" height="100%" width="100%" id="canvasGrid">
+   <rect fill="url(#gridpattern)" stroke-width="0" y="0" x="0" height="100%" width="100%"/>
+  </g>
+ </g>
+ <g>
+  <title>Layer 1</title>
+  <!--<rect rx="6" id="svg_1" height="95" width="110.5" y="7" x="7" stroke-width="11" stroke="#ffffff" fill="none"/>-->
```
Comment (ErikBoesen) [inline img/ropeclimb.svg]: More unnecessary content here.

File: index.html [hunk 5]
```diff
@@ -3,19 +3,25 @@
...
+    <title>VictiScout</title>
     <link rel="stylesheet" href="style.css">
 </head>
 
 <body>
     <div class="row">
         <input type="number" id="team" placeholder="Team #" class="large">
-        <input type="number" id="match" placeholder="Match #" class="special large">
+        <input type="number" id="match" placeholder="Match #" class="large">
+        <p>Playoffs<input type="checkbox" id="playoffs"></p>
```
Comment (ErikBoesen) [inline index.html]: Is this getting reset? It probably shouldn't.

File: index.html [hunk 6]
```diff
@@ -26,149 +32,107 @@ <h2 id="path-label">Save location:</h2>
...
+            <img src="img/gear.svg">
+            <button class="decrease">-</button>
+            <input type="number" id="gear" value="0" min="0">
+            <button class="increase">+</button>
+        </div>
+        <div class="tasks">
+            <h2>Rope Climb</h2>
+            <img src="img/ropeclimb.svg" id="ropeImage" onclick="ropeImage.onclick();">
+            <!-- TODO: Re-add attempted? -->
+            <p>Achieved? <input type="checkbox" id="rope-achieved" onclick="ropeChanged();"></p>
```
Comment (ErikBoesen) [inline index.html]: Using `onclick` properties in HTML is highly discouraged.

File: renderData.js [hunk 7]
```diff
@@ -3,37 +3,43 @@ const unirest = require('unirest');
...
 
+
     // Make column headers.
     // Create <tr> element to put everything in.
     var tr = document.createElement('tr');
     // Go through the first data object
     for (var j in data[0]) {
+        j = j.replace(/-/, ' ');
+        j = j.replace(/-/, ' ');
+        j = j.replace(/-/, ' ');
```
Comment (ErikBoesen) [inline renderData.js]: No.

File: renderData.js [hunk 8]
```diff
@@ -54,10 +60,171 @@ function render(data) {
...
+
+/*
+    var dict = [];
+    var minorDict = [];
+    //var table = document.getElementById('t')
+    //var rows = table.getElementsByTagName('tbody')[0].getElementsByTagName('tr');
+    var lowboilers = [];
+    /*for (var i; i<=rows.length; i++){
+        //console.log(i);
+        if (i != 0){
```
Comment (ErikBoesen) [inline renderData.js]: Remote this

File: renderData.js [hunk 9]
```diff
@@ -54,10 +60,171 @@ function render(data) {
...
+        majorDict.push([sum, dict[i][1]]);
+        minorDict.push(majorDict[i][1])
+    }
+    var superDict = [];
+    var subDict = [];
+    for (var i = 0; i<=minorDict.length; i++){
+        if (i%20 == 0 && i != 0){
+            superDict.push(subDict);
+            subDict = [];
+            subDict.push(minorDict[i])
```
Comment (ErikBoesen) [inline renderData.js]: Delete this commented out code.

File: script.js [hunk 10]
```diff
@@ -3,20 +3,25 @@ const fs = require('fs');
...
+	  reset: document.getElementById('reset'),
+	  view: document.getElementById('view'),
+    pathLabel: document.getElementById('path-label'),
+	  path: document.getElementById('path'),
+    autoAccuracy: document.getElementById('calculate-accuracy'),
+    teleopAccuracy: document.getElementById('teleop-calculate-accuracy'),
+    total: document.getElementById('calculate-total'),
+	  pathWarning: document.getElementById('path-warning'),
+    ropeAchieved: document.getElementById('rope-achieved'),
+    ropeImage: document.getElementById('ropeImage')}
```
Comment (ErikBoesen) [inline script.js]: We're specially defining a LOT of elements here.

File: script.js [hunk 11]
```diff
@@ -25,10 +30,8 @@ var tags = document.querySelectorAll('input:not(.special), select:not(.special)'
 var inputs = {};
 // Make each element be the value to a key named after its ID.
 for (i = 0; i < tags.length; i++) inputs[tags[i].id] = tags[i];
-
-
 // Submit data (also resets all fields).
-submit.onclick = function() {
+elements.submit.onclick = function() {
```
Comment (ErikBoesen) [inline script.js]: Not sure if `elements` is a great name. Would be better shorter; something like `pg`.

File: script.js [hunk 12]
```diff
@@ -134,24 +135,136 @@ function resetInputs() {
...
+	localStorage.path = elements.path.value;
+  localStorage.target = target.value;
 	// Tell main.js to open rendered data window
 	ipc.send('renderData');
 };
 
+// Begin the score processing stuff
+// Function for determining accuracy in autonomous
+elements.autoAccuracy.onclick = function() {
+    var autoMisses = document.getElementById('auto-high-boiler-misses').value;
```
Comment (ErikBoesen) [inline script.js]: This feels extremely messy. Plus, I really don't like the percentage measurements.

File: script.js [hunk 13]
```diff
@@ -134,24 +135,136 @@ function resetInputs() {
...
+    }
+    teleopAccuracy = Math.round(teleopAccuracy * 100) + '%';
+    document.getElementById('teleop-accuracy').value = teleopAccuracy;
+}
+// Change the image color and the checkbox value
+elements.ropeImage.onclick = function() {
+    //If the image is red, make it white, if it is white, make it red
+    if (elements.ropeImage.src.endsWith('img/ropeclimb.svg')) {
+        elements.ropeAchieved.checked=true;
+        elements.ropeImage.src = 'img/ropeclimb-red.svg';
```
Comment (ErikBoesen) [inline script.js]: Having an entirely duplicate image save one color is not a very good solution.

File: script.js [hunk 14]
```diff
@@ -134,24 +135,136 @@ function resetInputs() {
...
+    } else {
+        elements.ropeAchieved.checked=false;
+        elements.ropeImage.src = 'img/ropeclimb.svg';
+    }
+}
+// Link the checkbox onchange function (called in the document) to the image onclick
+ropeChanged = function() { elements.ropeImage.onclick(); };
```
Comment (ErikBoesen) [inline script.js]: NO.

File: script.js [hunk 15]
```diff
@@ -134,24 +135,136 @@ function resetInputs() {
...
+  // Return the values as an object
+  return {points: {total: totalPoints, auto: autoPoints, teleop: teleopPoints}, rotors: {total: rotors, auto: autoRotors, teleop: teleopRotors}, rankingPoints: rankingBonus}
+}
+// Totals up all points for the whole thing
+elements.total.onclick = function() {
+    // Call accuracy functions
+    elements.autoAccuracy.onclick();
+    elements.teleopAccuracy.onclick();
+    // Call steam function
+    var steamPoints = elements.steam.onclick();
```
Comment (ErikBoesen) [inline script.js]: Please, don't directly call `onclick` methods.

File: script.js [hunk 16]
```diff
@@ -134,24 +135,136 @@ function resetInputs() {
...
+    // The score depends on the game stage: in the playoffs, no ranking points are scored;
+    // In the qualifiers, only ranking points are scored (the match points still matter
+    // as wins/ties win ranking points)
+    var playoffs = document.getElementById('playoffs').checked;
+    // You get a ranking point from scoring 40kPa in auto, but in playoffs you get 20 points
+    var steamBonus = (steamPoints>40&&playoffs) ? 20 : 0;
+    // Call the rotorCalc function to calculate points (and ranking points) from gears
+    var rotorInfo = rotorCalc(autoGears, teleGears, playoffs);
+    // We only need the points right now
+    var rotorPoints = rotorInfo.points.total;
```
Comment (ErikBoesen) [inline script.js]: So many variables...

File: script.js [hunk 17]
```diff
@@ -134,24 +135,136 @@ function resetInputs() {
...
+    // Tally up all the points
+    var totalPoints = steamPoints + rotorPoints + ropePoints + steamBonus;
+    // Get ranking point values from the rotors and steam bonuses
+    var rankingPoints = rotorInfo.rankingPoints + ((steamPoints>40&&!playoffs) ? 1 : 0);
+    // Display the values for ranking points and total points
+    document.getElementById('ranking-points').value = rankingPoints;
+    document.getElementById('total-points').value = totalPoints;
+}
+// Calculate the steam produced (equivalent to 1 match point)
+elements.steam.onclick = function() {
```
Comment (ErikBoesen) [inline script.js]: I really don't think we need these readouts.

**Comment by Twinters007** [comment]
Yeah don't make a PR from  with 2 month old code

**Pull request closed without merging**